packages/browser-destinations/destinations/rupt/src/attach/index.ts

```diff
@@ -0,0 +1,81 @@
+import { BrowserActionDefinition } from '@segment/browser-destination-runtime/types'
+import { Payload } from 'src/attach.types'
+import { Settings } from 'src/generated-types'
+import Rupt from 'src/types'
+
+const attach: BrowserActionDefinition<Settings, Rupt, Payload> = {
+  title: 'Attach Device',
+  description: 'Attach a device to an account.',
+  defaultSubscription: 'type = "page"',
+  platform: 'web',
+  fields: {
+    account: {
+      description: 'The account to attach the device to.',
+      label: 'Account',
+      type: 'string',
+      required: true,
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    email: {
+      description: 'The email of the user to attach the device to.',
+      label: 'Email',
+      type: 'string',
+      required: false,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.email' },
+          then: { '@path': '$.traits.email' },
+          else: { '@path': '$.email' }
+        }
+      }
```

Comment:
```suggestion
      default: {
        '@if': {
          exists: { '@path': '$.context.traits.email' },
          then: { '@path': '$.context.traits.email' },
          else: { '@path': '$.properties.email' }
        }
      }
```

If you decide to keep the defaultSubscription as "type = 'page'", then you will only really see email addresses being send via context.traits.email or properties.email. 

traits.email is only really used in identify() payloads.